### DISCUSSION TITLE
Influxdb doesn't work with *.influxdata.com (needs https)

### DISCUSSION THREAD

**Discussion by DiegoB80** [arendst/Tasmota#13431] [Ideas]
### PROBLEM DESCRIPTION
I tried to connect tasmota v10.0.0 (build customized with influxdb) to influxdata.com (a cloud server that accepts only https connections).
It doesn't work because the request to the server is done over http (instead https).

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Basic
- [x] Tasmota binary firmware version number used: 10.0.0
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: Flashed Tasmota 9.2.0 via tasmotizer using usb-ttl adaptor, updgrade to Tasmota 10.0.0 via web UI
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua

14:57:06.577 CMD: Backlog Template; Module; GPIO 255
14:57:06.579 SRC: WebConsole from 192.168.1.1
14:57:06.582 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 26, Data 'Template; Module; GPIO 255'
14:57:06.652 SRC: Backlog
14:57:06.654 CMD: Grp 0, Cmnd 'TEMPLATE', Idx 1, Len 0, Data ''
14:57:06.659 RSL: RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
14:57:06.865 SRC: Backlog
14:57:06.867 CMD: Grp 0, Cmnd 'MODULE', Idx 1, Len 0, Data ''
14:57:06.871 RSL: RESULT = {"Module":{"1":"Sonoff Basic"}}
14:57:07.115 SRC: Backlog
14:57:07.118 CMD: Grp 0, Cmnd 'GPIO', Idx 1, Len 3, Data '255'
14:57:07.125 RSL: RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}



```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
15:00:44.051 CMD: ifxhost 3.123.149.45
15:00:44.053 SRC: WebConsole from 192.168.1.1
15:00:44.056 CMD: Grp 0, Cmnd 'IFXHOST', Idx 1, Len 12, Data '3.123.149.45'
15:00:44.060 CFG: CR 539/699, Busy 0
15:00:44.063 RSL: RESULT = {"IfxHost":"3.123.149.45"}
15:00:45.936 IFX: Url http://3.123.149.45:443/api/v2/write?org=Bortolot&bucket=MyHome
15:00:45.938 IFX: Validating connection to http://3.123.149.45:443/health
15:00:46.008 IFX: Error <html>
<head><title>400 The plain HTTP request was sent to HTTPS port</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<center>The plain HTTP request was sent to HTTPS port</center>
<hr><center>nginx</center>
</body>
</html>

```

### TO REPRODUCE
compile tasmota 10.0.0 with influxdb features
set the parameters of the cloud influxdb of your database
RESULT = {"Ifx":{"State":"ON","Host":"3.123.149.45","Port":443,"Version":2,"Bucket":"MyHome","Org":"Bortolot"}}
ifx 1
ifxhost "your host"
ifxBucket "your backet"
IfxToken "your token"
ifxOrg "your org"

PS: If you need I can generate a temporary bucket on my cloud in order to help the developer to fix the problem

PS: standard port doesn't work for cloud server. You can try with a telnet on IP: 3.123.149.45
I'm using this bucket with node-red and it works (with port 443)

### EXPECTED BEHAVIOUR
No connection error

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by arendst** [comment]
The current implementation does support http only.

**Comment by barbudor** [comment]
@DiegoB80 
~~In addition to not supporting https, the current influxdb driver is based on line-protocol which is supported only until influxdb v1.8
The InfluxDB Cloud SaaS is based on v2.0 which do not support the line-protocol.
So even with https, the current driver would not work with the cloud version~~

My mistake, the driver seems to support v2 as well. So only https is the blocking point

Are you using ESP8266 or ESP32 ?
On ESP32 Berry scripting language on Tasmota32 can support HTTPS and it may be possible to implement a driver for InfluxDB v2.0
If you are familiar with Python, Berry has a syntax and concepts very close and we can support you.

**Comment by DiegoB80** [comment]
@barbudor 

Thanks for your answer.
Good news if v2.0 supports the line-protocol!
I'm using ESP8266 (sonoff basic / dual and some devices like ESP01/ESP12F...). No experiences with ESP32 so far.
This feature is not mandatory for me, I'm using an 'on-premise' Raspberry pi to elaborate data before send to cloud.
I was just thinking that since Tasmota is getting really powerful (now I can manage devices via AWS or Azure IoT) if you add also this capability it's possible to remove the Raspberry in many projects. 
Unfortunately I am not familiar with python.
